http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -354,17 +354,33 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         F.blocking(wsmApi.getAzureManagedIdentity(workspaceId.value, wsmIdentity.resourceId.value))
       }
 
+      // Create or fetch WSM managed identity if shared app
+      wsmResourceApi <- buildWsmResourceApiClient
       // Call WSM to get the list of databases for the app.
-      wsmDatabases <- appControlledResourceQuery
-        .getAllForAppByType(app.id.id, WsmResourceType.AzureDatabase)
-        .transaction
-      wsmDatabases <- wsmDatabases.traverse { wsmDatabase =>
-        F.blocking(wsmApi.getAzureDatabase(workspaceId.value, wsmDatabase.resourceId.value))
-          .map(db => WsmControlledDatabaseResource(db.getMetadata.getName, db.getAttributes.getDatabaseName))
+      namespacePrefix = app.appResources.namespace.value
+      wsmManagedIdentityOpt <- app.samResourceId.resourceType match {
+        case SamResourceType.SharedApp =>
+          // if a managed identity has already been created in the workspace use that otherwise create a new managed identity
+          retrieveWsmManagedIdentity(wsmResourceApi, app.appType, workspaceId.value).flatMap {
+            case Some(v) => F.pure(Option(v))
+            case None    => createAzureManagedIdentity(app, namespacePrefix, workspaceId)
+          }
+        case _ => F.pure(None)
       }
 
+      // Create or fetch WSM databases
+      wsmDatabases <-
+        createOrFetchWsmDatabaseResources(
```

Comment:
A similar comment here -- I think we need `createOrFetch` semantics app _creation_ (since it may be a fresh workspace or a cloned workspace). But app _updates_ should never create a database, we should assume it already exists.

Reply:
createOrFetch will create them on app creation, and fetch them on app updates. But as I mentioned, I can separate this call into fetch only for updates.